Tests/Packet++Test/Tests/IcmpV6Tests.cpp

```diff
@@ -0,0 +1,287 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "EthLayer.h"
+#include "IPv6Layer.h"
+#include "IcmpV6Layer.h"
+#include "Logger.h"
+#include "MacAddress.h"
+#include "NdpLayer.h"
+#include "Packet.h"
+#include "SystemUtils.h"
+#include <sstream>
+
+PTF_TEST_CASE(IcmpV6ParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/IcmpV6_EchoRequest.dat");
+	READ_FILE_AND_CREATE_PACKET(2, "PacketExamples/IcmpV6_EchoReply.dat");
+	READ_FILE_AND_CREATE_PACKET(3, "PacketExamples/IcmpV6_NeighSoli.dat");
+	READ_FILE_AND_CREATE_PACKET(4, "PacketExamples/IcmpV6_NeighAdv.dat");
+	READ_FILE_AND_CREATE_PACKET(5, "PacketExamples/IcmpV6_NeighAdvNoOption.dat");
+
+	pcpp::Packet echoRequestPacket(&rawPacket1);
+	pcpp::Packet echoReplyPacket(&rawPacket2);
+	pcpp::Packet neighSoliPacket(&rawPacket3);
+	pcpp::Packet neighAdvPacket(&rawPacket4);
+	pcpp::Packet neighAdvPacketNoOpt(&rawPacket5);
+
+	// Echo request
+	PTF_ASSERT_TRUE(echoRequestPacket.isPacketOfType(pcpp::ICMPv6));
+	pcpp::ICMPv6EchoLayer *echoRequestLayer = echoRequestPacket.getLayerOfType<pcpp::ICMPv6EchoLayer>();
+	PTF_ASSERT_NOT_NULL(echoRequestLayer);
+	PTF_ASSERT_TRUE(echoRequestLayer->isMessageOfType(pcpp::ICMPv6MessageType::ICMPv6_ECHO_REQUEST));
+	PTF_ASSERT_EQUAL((int)echoRequestLayer->getMessageType(), 128);
```

Comment:
`getMessageType()` returns an enum, why casing to an `int`? 🤔 
Same in lines 51, 65, 84, 93

Reply:
currently i don't know how to handle the enum class together with PTF_ASSERT_EQUAL

PTF_ASSERT_EQUAL(echoRequestLayer->getMessageType(), pcpp::ICMPv6MessageType::ICMPv6_ECHO_REQUEST);
results in the error: error C2679: binary '<<': no operator found which takes a right-hand operand of type 'pcpp::ICMPv6MessageType'

same result when typing 'enum' as third argument

when i use 'enum class' as third argument, it compiles and runs correctly, but i'm a little bit confused about this... __VA_ARGS__##_PTF_PRINT_TYPE(actual) will be replaced by the precompiler to enum class_PTF_PRINT_TYPE(actual) where i currently dont know why this compiles?

what will be the correct solution?

Reply:
you're right, I'm sorry. I don't think `PTF_ASSERT_EQUAL` works with `enum class`. I'll fix that, and for now let's keep the existing cast to `int`